dhall/src/Dhall/Binary.hs

```diff
@@ -810,47 +810,19 @@ decode (TList (TInt 25 : xs)) = do
             empty
 
     process xs
-decode (TList [ TInt 26, t₁, _T₁ ]) = do
-    t₀  <- decode t₁
-    _T₀ <- decode _T₁
+decodeMaybe (TList [ TInt 26, t₁, _T₁ ]) = do
+    t₀  <- decodeMaybe t₁
+    _T₀ <- decodeMaybe _T₁
     return (Annot t₀ _T₀)
-decode _ =
+decodeMaybe _ =
     empty
 
-{-| Decode a Dhall expression
+decode :: Term -> Either DecodingFailure (Expr s Import)
+decode term = case decodeMaybe term of
+    Nothing         -> Left (CBORIsNotDhall term)
+    Just expression -> Right expression
```

Comment:
I wonder if we could we provide backwards-compatibility in some way here?

E.g. something like
```haskell
decode :: Term -> Either DecodingFailure (Expr s Import)
decode term = case decodeMaybe term of
    Just expression -> Right expression
    Nothing -> case term of
        -- Pre-6.0.0 expressions
        TList [ TString version, subTerm ] -> case decodeMaybe subTerm of
          Just expression -> Right expression
          Nothing -> Left (CBORIsNotDhall subTerm)
        _ -> Left (CBORIsNotDhall term)
```

The reason is that otherwise every user will have to reimplement some kind of compatibility layer (and this means "forking" `Dhall.Import.exprFromImport` to have it call this code basically)